site/src/pages/[shortname]/[blockSlug].page.tsx

```diff
@@ -110,28 +110,43 @@ export const getStaticProps: GetStaticProps<
 > = async ({ params }) => {
   const { shortname, blockSlug } = parseQueryParams(params || {});
 
+  if (!shortname.startsWith("@")) {
+    return { notFound: true };
+  }
+
   const catalog = readBlocksFromDisk();
 
-  const metadata: BlockMetadata = await fetch(
+  const blockMetadataResponse = await fetch(
     `${BASE_URL}/blocks/${shortname}/${blockSlug}/metadata.json`,
-  ).then((res) => res.json());
+  );
+
+  if (blockMetadataResponse.status === 404) {
+    // TODO: Render custom 404 page for blocks
+    return { notFound: true };
+  } else if (!blockMetadataResponse.ok) {
+    // TODO: Render details of upstream error
+    throw new Error("Something went wrong");
```

Comment:
Thanks. Wouldn't this cause the page to crash? Perhaps we still return `{notFound: true}`, then add a comment to log the error

Reply:
Yeah that will cause the page to crash but that's deliberate. We don't want to mask network or parse errors with a 404. Ideally, we'll need to convert the error into some props and render a nice error page, but just throwing 500 looks more honest than showing 404 for a block that does exist.

Reply:
My current goal is to fix errors in paths like https://blockprotocol.org/assets/bp_twitter_cover.png_oops